tests/baselines/reference/parenthesizedContexualTyping1.types

```diff
@@ -220,11 +220,11 @@ var l = fun(((Math.random() < 0.5 ? ((x => x)) : ((x => undefined)))), ((x => x)
 >x => undefined : (x: number) => any
 >x : number
 >undefined : undefined
->((x => x)) : (x: number) => number
->(x => x) : (x: number) => number
->x => x : (x: number) => number
->x : number
->x : number
+>((x => x)) : (x: any) => any
+>(x => x) : (x: any) => any
+>x => x : (x: any) => any
+>x : any
+>x : any

```

Comment:
Ah I see. That is really complicated and very hard for anybody to understand. I definitely think we need to make the inference context shared, and make it smarter. It has to distinguish between type parameters that were fixed because of contextual typing, versus type parameters that were just finalized because the inference round was over.